frontends/parsers/parserDriver.cpp

```diff
@@ -0,0 +1,238 @@
+#include "parserDriver.h"
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/format.hpp>
+
+#include <cerrno>
+#include <cstdio>
+#include <iostream>
+#include <sstream>
+
+#include "frontends/common/constantFolding.h"
+#include "frontends/parsers/p4/p4lexer.hpp"
+#include "frontends/parsers/p4/p4parser.hpp"
+#include "frontends/parsers/v1/v1lexer.hpp"
+#include "frontends/parsers/v1/v1parser.hpp"
+#include "lib/error.h"
+
+
+#ifdef HAVE_LIBBOOST_IOSTREAMS
+
+#include <boost/iostreams/device/file_descriptor.hpp>
+#include <boost/iostreams/stream.hpp>
+
+namespace {
+
+/// A RAII helper class that provides an istream wrapper for a stdio FILE*. This
+/// is the efficient implementation for users with boost::iostreams installed.
+struct AutoStdioInputStream {
+    explicit AutoStdioInputStream(FILE* in)
```

Comment:
I'm not aware of an iostreams-based alternative to popen, unfortunately, though that would be ideal. My plan in the long term was to eliminate the need for `FILE` support by implementing the C preprocessor features we need internally.